Proton/Sources/Swift/Core/LayoutManager.swift

```diff
@@ -243,6 +244,22 @@ class LayoutManager: NSLayoutManager {
         return image
     }
 
+    private func resizedImage(image: UIImage, size: CGSize) -> UIImage {
+        let result: UIImage
+
+        let sizeCache = imagecache[image, default: .init()]
+        imagecache.updateValue(sizeCache, forKey: image)
+
+        let key = size.debugDescription as NSString
```

Comment:
I'm open to suggestion on how to further optimize the keying strategy. With this change, the most expensive operation becomes generating the string description for the size (the whole 4ms in the screenshot above). Interpolating the image and size in a single string was coming up at 11ms, which is still better than the original 650ms, but I ended choosing the fastest version.

Reply:
this seems pretty good @erudel. We can evaluate if we can use a better strategy than this for generating the key but this in itself seems like a substantial improvement. Thank you for taking this up.